src/session/session_api.c

```diff
@@ -1475,6 +1476,24 @@ __wt_session_range_truncate(
               session, EINVAL, "the start cursor position is after the stop cursor position");
     }
 
+    /*
+     * Utilize the cursor lower bound as the temporary buffer to storing the original start and stop
+     * key. We use the original keys for the write-ahead log.
+     */
+    if (start != NULL) {
+        if (WT_CURSOR_BOUNDS_SET(start))
+            WT_ERR_MSG(session, EINVAL, "cursor with bounds is not compatible with truncate operation.");
```

Comment:
Is this change related to this ticket or is it really a bounded cursor change that should be on its own? Should a new ticket be opened and this change merged on its own? (Same with the related check on the stop cursor.)

Reply:
As far as I understand this isn't directly related to the bounded cursor work. The two WT_ITEMs added to the session for the bounded cursor are being co-opted for WT-9715 to log the truncate start and stop keys.